app/models/ont/run.rb

```diff
@@ -77,29 +81,60 @@ def cancel
     end
 
     def flowcell_attributes=(flowcell_options)
-      self.flowcells = flowcell_options.map do |attributes|
+      transform_flowcell_attributes(flowcell_options)
+
+      options_ids = flowcell_options.pluck(:id).compact
+      flowcells.each do |flowcell|
+        flowcells.delete(flowcell) if options_ids.exclude? flowcell.id
+      end
+
+      flowcell_options.map do |attributes|
         if attributes[:id]
           update_flowcell(attributes)
         else
-          Ont::Flowcell.new(attributes)
+          create_flowcell(attributes)
         end
       end
     end
 
+    # Returns error messages added so far
+    def errors_messages
```

Comment:
Can you use errors.full_messages?

Reply:
I have removed the redundant method and duplicate message checking.